core/server/common/src/test/java/alluxio/master/journal/raft/transport/CopycatGrpcTransportTest.java

```diff
@@ -0,0 +1,346 @@
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.raft.transport;
+
+import alluxio.conf.ServerConfiguration;
+import alluxio.master.journal.raft.RaftJournalSystem;
+import alluxio.security.user.ServerUserState;
+
+import io.atomix.catalyst.buffer.BufferInput;
+import io.atomix.catalyst.buffer.BufferOutput;
+import io.atomix.catalyst.concurrent.SingleThreadContext;
+import io.atomix.catalyst.concurrent.ThreadContext;
+import io.atomix.catalyst.serializer.CatalystSerializable;
+import io.atomix.catalyst.serializer.Serializer;
+import io.atomix.catalyst.transport.Address;
+import io.atomix.catalyst.transport.Client;
+import io.atomix.catalyst.transport.Connection;
+import io.atomix.catalyst.transport.Server;
+import io.atomix.catalyst.transport.Transport;
+import io.atomix.copycat.protocol.ClientRequestTypeResolver;
+import io.atomix.copycat.protocol.ClientResponseTypeResolver;
+import io.atomix.copycat.server.storage.util.StorageSerialization;
+import io.atomix.copycat.server.util.ServerSerialization;
+import io.atomix.copycat.util.ProtocolSerialization;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+/**
+ * Units tests for {@link CopycatGrpcTransport}.
+ */
+public class CopycatGrpcTransportTest {
+
+  private Transport mTransport;
+
+  @Before
+  public void before() {
+    mTransport = new CopycatGrpcTransport(ServerConfiguration.global(), ServerUserState.global());
+  }
+
+  @After
+  public void after() {
+    mTransport.close();
+  }
+
+  @Test
+  public void testEstablishConnection() throws Exception {
+    // Set by server connection listener, when new connection is opened to server.
+    final AtomicBoolean connectionEstablished = new AtomicBoolean(false);
+    // Server connection lister that validates connection establishment.
+    Consumer<Connection> connectionListener = new CopycatTransportTestListener((connection) -> {
+      // Set connection as established.
+      connectionEstablished.set(true);
+    });
+
+    // Catalyst thread context for managing client/server.
+    ThreadContext connectionContext = createSingleThreadContext("ClientServerCtx");
+
+    // Address for binding transport server.
+    Address address = new Address("localhost", 10001);
```

Comment:
Should we registered these with `ReservedPorts`?

Reply:
`PortCoordination` and `ReservedPort` utilities are for minicluster tests.
Changed static ports to binding dynamically.